### PR TITLE
fix: enable live reloading title of doc

### PR DESCRIPTION
**Pull request by tusharf5**
## Motivation

Fix enable live reloading when doc title is changed [#1213](https://github.com/facebook/Docusaurus/issues/1213)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Local Testing. The blog is working locally after changes without any errors or warnings.

[Demo Video Before Changes](https://drive.google.com/file/d/1OasLEv3e8-Vk9ZrnaOjOylj1OejR7z_j/view?usp=sharing)

[Demo Video After Changes](https://drive.google.com/file/d/1XfF4RnpI-bHLNkPBbsqltNebyL_o3hb4/view?usp=sharing)

## Related PRs

N.A

### CODE REVIEW

**Comment by tusharf5** [comment]
> Please rebase. And also, don't just reload when title is changed. I think we should also reload for other metadata cases :)
> 
> For example
> https://docusaurus.io/docs/en/doc-markdown#markdown-headers
> 
> hide_title and sidebar_label

@endiliey  Made the changes 👍

**Comment by tusharf5** [comment]
Hey, @endiliey I just realized that I did a `git merge` instead of a `rebase`. Will it be an issue? If yes then I can re-submit the PR. My Bad, It's my first PR for an open source package 😥.

**Comment by yangshun** [comment]
@tusharf5 Merging is fine, the more important thing is you have the latest code within your branch.

File: packages/docusaurus-1.x/lib/server/server.js [hunk 1]
```diff
@@ -116,18 +120,35 @@ function execute(port, host) {
...
+    );
+
+    let metadata = Metadata[metakey];
 
     const file = docs.getFile(metadata);
     if (!file) {
       next();
       return;
     }
+    const extractedMetadata = metadataUtils.extractMetadata(file);
```
Comment (endiliey) [inline packages/docusaurus-1.x/lib/server/server.js]:
```js
const {rawContent, metadata: rawMetadata} = metadataUtils.extractMetadata(file):
```

File: packages/docusaurus-1.x/lib/server/server.js [hunk 2]
```diff
@@ -116,18 +120,35 @@ function execute(port, host) {
...
+    const rawContent = extractedMetadata.rawContent;
+    const rawMetadata = extractedMetadata.metadata;
+
+    // if any of the followings is changed, reload the metadata
+    const reloadTriggers = ['sidebar_label', 'hide_title', 'title'];
```
Comment (endiliey) [inline packages/docusaurus-1.x/lib/server/server.js]:
```js
if (reloadTriggers.some(key => metadata[key] !== rawMetadata[key]) {
  reloadMetadata();
  extractTranslations();
  reloadTranslations();
  metadata = Metadata[metakey];
}

**Comment by tusharf5** [comment]
@endiliey I've pushed the changes that you suggested. : )

**Comment by endiliey** [comment]
did u test it out again ? im on vacation so my response is slow

**Comment by tusharf5** [comment]
Oh okay : ) and yeah I've tested it out, it works as expected. (live reloads for the specified metadata attributes) @endiliey

**Comment by endiliey** [comment]
ok. lets ship it. 🔥🔥 u ready ? 👀

**Comment by tusharf5** [comment]
Thank you : )

**Comment by endiliey** [comment]
Nah, thanks to you as the contibutor😛👍

**Pull request merged**